### PR TITLE
Remove `constants.DisableRuntime` in favor of async config option.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2804" title="DX-2804" target="_blank">TaskDX-2804</a>  Pair async runtimes with ACTIVESTATE_CLI_DISABLE_RUNTIME
  </td></table>
  <br />

### CODE REVIEW

**Comment by mitchell-as** [comment]
The Windows test failures are due to timeouts and unrelated to this PR.

File: internal/runbits/runtime/requirements/requirements.go [hunk 1]
```diff
@@ -222,65 +221,63 @@ func (r *RequirementOperation) ExecuteRequirementOperation(ts *time.Time, requir
...
-		var trig trigger.Trigger
-		switch ns.Type() {
-		case model.NamespaceLanguage:
-			trig = trigger.TriggerLanguage
-		case model.NamespacePlatform:
-			trig = trigger.TriggerPlatform
-		default:
-			trig = trigger.TriggerPackage
-		}
+	ns := requirements[0].Namespace
```
Comment (mitchell-as) [inline internal/runbits/runtime/requirements/requirements.go:224]: Just removed the top-level `if` conditional. Nothing changed inside the block.

File: internal/runners/initialize/init.go [hunk 2]
```diff
@@ -288,31 +287,28 @@ func (r *Initialize) Run(params *RunParams) (rerr error) {
...
-			solveSpinner.Stop(locale.T("progress_fail"))
-			logging.Debug("Deleting remotely created project due to runtime setup error")
-			err2 := model.DeleteProject(namespace.Owner, namespace.Project, r.auth)
-			if err2 != nil {
-				multilog.Error("Error deleting remotely created project after runtime setup error: %v", errs.JoinMessage(err2))
-				return locale.WrapError(err, "err_init_refresh_delete_project", "Could not setup runtime after init, and could not delete newly created Platform project. Please delete it manually before trying again")
-			}
-			return errs.Wrap(err, "Failed to fetch build result")
+	// Solve runtime
+	solveSpinner := output.StartSpinner(r.out, locale.T("progress_solve"), constants.TerminalAnimationInterval)
```
Comment (mitchell-as) [inline internal/runners/initialize/init.go:291]: Again, just removed the top-level `if` conditional. Nothing changed inside the block.

File: test/integration/revert_int_test.go [hunk 3]
```diff
@@ -24,10 +24,7 @@ func (suite *RevertIntegrationTestSuite) TestRevert() {
 
 	// Revert the commit that added urllib3.
 	commitID := "1f4f4f7d-7883-400e-b2ad-a5803c018ecd"
-	cp := ts.SpawnWithOpts(
-		e2e.OptArgs("revert", commitID),
-		e2e.OptAppendEnv(constants.DisableRuntime+"=true"),
-	)
+	cp := ts.Spawn("revert", commitID)
```
Comment (MDrakos) [inline test/integration/revert_int_test.go:27]: Should we set the async runtime config value for this test?

File: test/integration/shells_int_test.go [hunk 4]
```diff
@@ -49,7 +48,7 @@ func (suite *ShellsIntegrationTestSuite) TestShells() {
 			}
 
 			// Run the checkout in a particular shell.
-			cp = ts.SpawnShellWithOpts(shell, e2e.OptAppendEnv(constants.DisableRuntime+"=true"))
+			cp = ts.SpawnShellWithOpts(shell)
```
Comment (MDrakos) [inline test/integration/shells_int_test.go:51]: Set async runtime?

File: test/integration/activate_int_test.go [hunk 5]
```diff
@@ -54,11 +54,7 @@ func (suite *ActivateIntegrationTestSuite) TestActivateWithoutRuntime() {
 	close := suite.addForegroundSvc(ts)
 	defer close()
 
-	cp := ts.SpawnWithOpts(
-		e2e.OptArgs("activate", "ActiveState-CLI/Empty"),
-		e2e.OptAppendEnv(constants.DisableRuntime+"=true"),
-	)
-	cp.Expect("Skipping runtime setup")
+	cp := ts.Spawn("activate", "ActiveState-CLI/Empty")
```
Comment (MDrakos) [inline test/integration/activate_int_test.go:57]: Set async runtime?

**Review by MDrakos — CHANGES_REQUESTED** [review]
Just a few tests to double check that we have the async runtimes setup.

File: test/integration/revert_int_test.go (hunk 3 above)
Comment (mitchell-as) [reply test/integration/revert_int_test.go:27]: No, we need it for the later use of `state shell`. Originally I enabled it, but that caused a failure and caused me to file another ticket and put up #3413.

File: test/integration/shells_int_test.go (hunk 4 above)
Comment (mitchell-as) [reply test/integration/shells_int_test.go:51]: Nope, we need it for the same reason as stated above.

File: test/integration/activate_int_test.go (hunk 5 above)
Comment (mitchell-as) [reply test/integration/activate_int_test.go:57]: No, this is the empty project and we don't care about it. This use of DisableRuntime is vestigal.

**Review by MDrakos — APPROVED** [review]
Thanks for double checking!

**Pull request merged**